### ISSUE TITLE
Text is converted to lower case when adding data to the database.

### ISSUE THREAD

**Issue by pafi-code** [weaviate/weaviate#4213]
### How to reproduce this bug?

So we started a container running weaviate via docker compose. We use the default vectorization module `text2vec-openai` with azure endpoint.'The compose file looks something like this

```
version: '3.4'
services:
  weaviate:
    command:
    - --host
    - 0.0.0.0
    - --port
    - '8080'
    - --scheme
    - http
    image: semitechnologies/weaviate:1.23.9
    ports:
    - 8080:8080
    - 50051:50051
    restart: on-failure:0
    environment:
      AZURE_APIKEY: ''
      QUERY_DEFAULTS_LIMIT: 25
      AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: 'true'
      PERSISTENCE_DATA_PATH: '/var/lib/weaviate'
      DEFAULT_VECTORIZER_MODULE: 'text2vec-openai'
      ENABLE_MODULES: 'text2vec-openai'
      CLUSTER_HOSTNAME: 'node1'
```

Now when I add my data to the database everything seems normal from the outside.

```python
import weaviate # weaviate-client==4.4.4, python==3.12.1
import weaviate.classes as wvc

config = wvc.config.Configure.Vectorizer.text2vec_azure_openai(
    base_url="your.url",
    resource_name="your.resource",
    deployment_id="weaviate-ada-002",
    vectorize_collection_name=False,
)

with weaviate.connect_to_local() as client:
    client.collections.create(
        name="test",
        description="some tests",
        vectorizer_config=config,
        properties=[
            wvc.config.Property(
                name="text",
                data_type=wvc.config.DataType.TEXT,
                description="The text.",
                vectorize_property_name=False,
            )
        ]
    )
    collection = client.collections.get("test")
    collection.data.insert(properties={"text": "This is a text"})
```

### What is the expected behavior?

I would expect that when I query the database for the text `This is a text` I would assume I would get the object with its corresponding embedding.

### What is the actual behavior?

I get the object with everything correct except for the embedding. The embedding shown is for the text `this is a text`.
Notice that everything was converted into lower.

### Supporting information

The issue seems to be in the go module [text2vec-openai](https://github.com/weaviate/weaviate/tree/master/modules/text2vec-openai). The request that is created looks like this
![Screenshot from 2024-02-14 14-35-54_hidden](https://github.com/weaviate/weaviate/assets/69895152/033caa3c-366e-499a-b958-5b9545d0d035)

### Server Version

1.23.9

### Code of Conduct

- [X] I have read and agree to the Weaviate's [Contributor Guide](https://weaviate.io/developers/contributor-guide) and [Code of Conduct](https://weaviate.io/service/code-of-conduct)

**Comment by litlig** [comment]
there was a related issue here https://github.com/weaviate/weaviate/issues/3644, we can make it a user config to choose whether or not to convert property values to lower case

**Comment by pafi-code** [comment]
Yea but even then it should be somewhere documented that when searching for `near_text` the input query is not converted to lowercase which makes it inconsistent.

**Comment by dirkkul** [comment]
will be fixed in 1.27: https://github.com/weaviate/weaviate/pull/5796
